client/my-sites/upgrades/domain-management/dns/dns-record.jsx

```diff
@@ -75,24 +77,45 @@ var DnsRecord = React.createClass( {
 	},
 
 	getName: function() {
-		var { name, service, protocol } = this.props.dnsRecord,
+		const { name, service, protocol, type } = this.props.dnsRecord,
 			domain = this.props.selectedDomainName,
 			isRoot = name === domain + '.';
 
-		if ( 'SRV' === this.props.dnsRecord.type ) {
+		if ( 'SRV' === type ) {
 			return service + '_' + protocol + '.' + ( isRoot ? name + '.' : '' ) + domain;
 		}
 
-		return isRoot ? domain : name + '.' + domain;
+		if ( endsWith( name, '.' ) ) {
+			return name.slice( 0, -1 );
+		}
```

Comment:
Is it possible to use the same approach in the normalize function above? I feel like it'd make it much simpler.